### ISSUE TITLE
Spurious error when hovering text

### ISSUE THREAD

**Issue by WilliamVenner** [rust-lang/rust-analyzer#12843]
Happens completely randomly, have been experiencing this for about a week now

```rs
version: 0.3.1131-standalone (897a7ec4b 2022-07-17)
request: textDocument/hover HoverParams {
    text_document: TextDocumentIdentifier {
        uri: Url {
            scheme: "file",
            cannot_be_a_base: false,
            username: "",
            password: None,
            host: None,
            port: None,
            path: "/c%3A/Users/William/Documents/GitHub/beam-tauri/daemon/src/network/discovery.rs",
            query: None,
            fragment: None,
        },
    },
    position: Position(
        Position {
            line: 121,
            character: 6,
        },
    ),
    work_done_progress_params: WorkDoneProgressParams {
        work_done_token: None,
    },
}

> impl_trait_query(ImplId(10977) -> ItemLoc { container: ModuleId { krate: CrateId(211), block: None, local_id: Idx::<ModuleData>(1) }, id: ItemTreeId { tree: TreeId { file: HirFileId(FileId(FileId(2757))), block: None }, value: Idx::<Impl>(0) } } -> ImplData { target_trait: None, self_ty: Path(Path { type_anchor: None, mod_path: ModPath { kind: Plain, segments: [Name(Text("RuntimeFlavor"))] }, generic_args: [None] }), items: [FunctionId(FunctionId(54766))], is_negative: false, attribute_calls: None })

> impl_self_ty_query(ImplId(10977) -> ItemLoc { container: ModuleId { krate: CrateId(211), block: None, local_id: Idx::<ModuleData>(1) }, id: ItemTreeId { tree: TreeId { file: HirFileId(FileId(FileId(2757))), block: None }, value: Idx::<Impl>(0) } } -> ImplData { target_trait: None, self_ty: Path(Path { type_anchor: None, mod_path: ModPath { kind: Plain, segments: [Name(Text("RuntimeFlavor"))] }, generic_args: [None] }), items: [FunctionId(FunctionId(54766))], is_negative: false, attribute_calls: None })

> DefMap CrateId(211) crate_name=Some(CrateDisplayName { crate_name: CrateName("tinyvec_macros"), canonical_name: "tinyvec_macros" }) block=None path=RuntimeFlavor

thread 'Worker' panicked at 'index out of bounds: the len is 1 but the index is 1', crates\hir-def\src\nameres\path_resolution.rs:381:33
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library\std\src\panicking.rs:584
   1: core::panicking::panic_fmt
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library\core\src\panicking.rs:142
   2: core::panicking::panic_bounds_check
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library\core\src\panicking.rs:84
   3: hir_def::nameres::path_resolution::<impl hir_def::nameres::DefMap>::resolve_path_fp_with_macro_single
   4: hir_def::resolver::Resolver::resolve_path_in_type_ns
   5: hir_ty::lower::TyLoweringContext::lower_ty_ext
   6: hir_ty::lower::impl_self_ty_query
   7: salsa::runtime::Runtime::execute_query_implementation
   8: core::ptr::drop_in_place<salsa::derived::slot::Memo<hir_ty::db::ProgramClausesForChalkEnvQuery>>
   9: salsa::derived::slot::Slot<Q,MP>::read
  10: <salsa::derived::DerivedStorage<Q,MP> as salsa::plumbing::QueryStorageOps<Q>>::try_fetch
  11: salsa::QueryTable<Q>::get
  12: <DB as hir_ty::db::HirDatabase>::impl_self_ty
  13: hir_ty::lower::impl_trait_query
  14: salsa::runtime::Runtime::execute_query_implementation
  15: core::ptr::drop_in_place<salsa::derived::slot::Memo<hir_ty::db::ProgramClausesForChalkEnvQuery>>
  16: salsa::derived::slot::Slot<Q,MP>::maybe_changed_since
  17: <salsa::derived::DerivedStorage<Q,MP> as salsa::plumbing::QueryStorageOps<Q>>::maybe_changed_since
  18: hir_ty::db::HirDatabaseGroupStorage__::maybe_changed_since
  19: <ide_db::RootDatabase as salsa::plumbing::DatabaseOps>::maybe_changed_since
  20: salsa::derived::slot::MemoRevisions::validate_memoized_value
  21: core::ptr::drop_in_place<salsa::derived::slot::Memo<hir_ty::db::ProgramClausesForChalkEnvQuery>>
  22: salsa::derived::slot::Slot<Q,MP>::read
  23: <salsa::derived::DerivedStorage<Q,MP> as salsa::plumbing::QueryStorageOps<Q>>::try_fetch
  24: <DB as hir_ty::db::HirDatabase>::trait_impls_in_crate
  25: hir::Impl::all_for_type
  26: core::ops::function::impls::<impl core::ops::function::FnMut<A> for &mut F>::call_mut
  27: hashbrown::raw::RawTable<T,A>::insert
  28: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
  29: ide::goto_implementation::goto_implementation
  30: std::panicking::try
  31: core::ops::function::impls::<impl core::ops::function::FnMut<A> for &mut F>::call_mut
  32: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
  33: rust_analyzer::handlers::handle_hover
  34: std::panicking::try
  35: <F as threadpool::FnBox>::call_box
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
[Error - 21:09:38] Request textDocument/hover failed.
  Message: request handler panicked: index out of bounds: the len is 1 but the index is 1
  Code: -32603 
```

**rust-analyzer version**: (eg. output of "Rust Analyzer: Show RA Version" command)
rust-analyzer version: 0.3.1131-standalone (897a7ec4b 2022-07-17)

**rustc version**: (eg. output of `rustc -V`)
rustc 1.62.0 (a8314ef7d 2022-06-27)

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTUP_HOME` or `CARGO_HOME`)
All default

**Comment by Veykril** [comment]
This is a known issue https://github.com/rust-lang/rust-analyzer/issues/10084

In short, this may happen when you edit the dependencies in one of your Cargo.toml files. Once you get this panic it will constantly repeat until your restart the rust-analyzer server.

**Comment by lnicola** [comment]
Do you see this after editing `Cargo.toml`? CC https://github.com/rust-lang/rust-analyzer/issues/12046.

**Comment by WilliamVenner** [comment]
I get this every day, but it only started about a week ago, so I don't know if it's the same issue.

**Comment by WilliamVenner** [comment]
More

```
version: 0.3.1131-standalone (897a7ec4b 2022-07-17)
request: textDocument/completion CompletionParams {
    text_document_position: TextDocumentPositionParams {
        text_document: TextDocumentIdentifier {
            uri: Url {
                scheme: "file",
                cannot_be_a_base: false,
                username: "",
                password: None,
                host: None,
                port: None,
                path: "/c%3A/Users/William/Documents/GitHub/beam-tauri2/daemon/src/input/mouse/mod.rs",
                query: None,
                fragment: None,
            },
        },
        position: Position {
            line: 7,
            character: 16,
        },
    },
    work_done_progress_params: WorkDoneProgressParams {
        work_done_token: None,
    },
    partial_result_params: PartialResultParams {
        partial_result_token: None,
    },
    context: Some(
        CompletionContext {
            trigger_kind: Invoked,
            trigger_character: None,
        },
    ),
}

thread 'Worker' panicked at 'assertion failed: start <= end', C:\Users\runneradmin\.cargo\registry\src\github.com-1ecc6299db9ec823\text-size-1.1.0\src\range.rs:48:9
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library\std\src\panicking.rs:584
   1: core::panicking::panic_fmt
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library\core\src\panicking.rs:142
   2: core::panicking::panic
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library\core\src\panicking.rs:48
   3: ide_completion::completions::item_list::trait_impl::complete_trait_impl_name
   4: ide_completion::completions
   5: std::panicking::try
   6: rust_analyzer::handlers::handle_completion
   7: std::panicking::try
   8: <F as threadpool::FnBox>::call_box
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
[Error - 22:03:37] Request textDocument/completion failed.
  Message: request handler panicked: assertion failed: start <= end
  Code: -32603 
```

**Comment by Veykril** [comment]
This is a different panic, could you make a new issue for that one?
